### PR TITLE
[ROCm] hipdnn integration batchnorm PR1

### PR DESCRIPTION
**Pull request by dnikolaev-amd**
## Motivation

Use HipDNN in Pytorch

## Technical Details

This is the first PR of two to comply with [Pytorch Forwards compatibility policy](https://github.com/pytorch/pytorch/wiki/PyTorch's-Python-Frontend-Backward-and-Forward-Compatibility-Policy#forwards-compatibility-fc)

Requirements:
- ROCm 7.12 or higher

What was done:
- Find and integrate HipDNN with Pytorch if available (cmake, includes, libs)
- Cmake definition `USE_HIPDNN` if HipDNN is available
- Runtime hook `CUDAHooks::compiledWithHipDNN()`  
- HipDNN settings in `globalContext()` (hipdnn is disabled by default now)
- Add `BatchNormBackend::Hipdnn` into batcnorm backend selection logic
- Hipdnn batchnorm implementation
- Add hipdnn batchnorm to `native_functions.yaml` and `derivatives.yaml`

HipDNN backend is disabled by default. There are no ways for user to enable HipDNN in python. PR2 for the python interface will follow in two weeks per FC policy 

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang

### CODE REVIEW

**Comment by dnikolaev-amd** [comment]
Another extended PR https://github.com/pytorch/pytorch/pull/177534
